src/JWT/JwtBuilder.cs

```diff
@@ -0,0 +1,266 @@
+        /// <param name="value">The value you want give to the header.</param>
+        /// <returns>The current builder-instance</returns>
+        public JwtBuilder AddHeader(HeaderName name, string value)
+        {
+            _jwt.Header.Add(name.GetHeaderName(), value);
+            return this;
+        }
+
+        /// <summary>
+        /// Add any claim you want to the JWT
+        /// </summary>
+        /// <param name="name">Your name of the Claim.</param>
+        /// <param name="value">Your value of the Claim. It will be parse to JSON.</param>
+        /// <returns>The current builder-instance</returns>
+        public JwtBuilder AddClaim(string name, object value)
+        {
+            _jwt.Payload.Add(name, value);
+            return this;
+        }
+
+        /// <summary>
+        /// Add string claims to the JWT payload.
+        /// </summary>
+        /// <param name="name">Your name of the Claim.</param>
+        /// <param name="value">Your value of the claim as String.</param>
+        /// <returns>The current builder-instance</returns>
+        public JwtBuilder AddClaim(string name, string value) => this.AddClaim(name, (object)value);
+        /// <summary>
+        /// Add public claims to the JWT payload.
+        /// </summary>
+        /// <param name="names">The name of the public claim you want set</param>
+        /// <param name="value">The string-value for the public claim</param>
+        /// <returns>The current builder-instance</returns>
+        public JwtBuilder AddClaim(PublicClaimsNames names, string value) => this.AddClaim(names.GetPublicClaimName(), value);
+        /// <summary>
+        /// Set a custom Serializier. If you don't set this it will be <see cref="JsonNetSerializer" /> use.
+        /// </summary>
+        /// <param name="serializer">The serializier instance you want use.</param>
+        /// <returns>The current builder-instance</returns>
+        public JwtBuilder SetSerializer(IJsonSerializer serializer)
+        {
+            _serializer = serializer;
+            return this;
+        }
+
+        /// <summary>
+        /// Set a time provieder to get the time reference.!--
+        /// If you don't call this the builder will use <see cref="UtcDateTimeProvider"/>
+        /// </summary>
+        /// <param name="provider">Zou custom provider you want use.</param>
+        /// <returns>The current builder-instance</returns>
+        public JwtBuilder SetTimeProvider(IDateTimeProvider provider)
+        {
+            _utcProvieder = provider;
+            return this;
+        }
+
+        /// <summary>
+        /// Set the validTor. This is requierd for decode with verification!
+        /// </summary>
+        /// <param name="validTor">Your custom validTor</param>
+        /// <returns>The current builder-instance</returns>        
+        public JwtBuilder SetValidTor(IJwtValidTor validTor)
```

Comment:
`SetValidator`

Reply:
Don't know how this happened, was replaced everywhere. Reverted back.